### DISCUSSION TITLE
Bluetooth speaker disconnects after playing for a couple of minutes

### DISCUSSION THREAD

**Discussion by birkb85** [rern/rAudio#1297] [General]
Hi,

Its my first time asking a question here, and English is not my native language, please be friendly. :)

I have a problem where my bluetooth speaker disconnects after playing for 1 to about 12 minutes..
The speaker disconnects from the RPi, and the music is paused.
I am then able to go to Network settings and reconnect to the speaker and then click play to resume playing the music.
I cannot really find a pattern for why it happens..

I am using a RPi 3B, running i20230212 with updates installed.
I have connected:
* TP-Link UB500 Bluetooth dongle and disabled the on board bluetooth.
* 128GB USB dongle for music storage.
* The 7" official touch screen display.
* The RPi is connected to the internet via wifi.
* The Bluetooth speaker is an Anker Soundcore 3.
* Its running on an official 2A power supply, which powers it all.

I have tried almost all that I could think of to find the problem, but nothing so far solves it..
I have tested:
* The bluetooth dongle seems to stay connected to the speaker and playing fine while the dongle is connected to my laptop, so I think the dongle is ok.
* The speaker plays fine when connected to the phone, so I think the speaker is also ok.
* I thought maybe the power supply was not able to power it all, I have tried connecting a supply to the 7" display directly, but did not seem to change anything.
* I have tried connecting the RPi to Lan and disable wifi, but that does not seem to make a difference either.

I have thought of trying to use a powered USB hub if the problem is that the USB is not giving enough power, but do not own one, so have not tested this yet.

Do you have any suggestions on what might be the problem?

Best regards

**Comment by birkb85** [comment]
Hmm, got a little further.
When the RPi is connected to Lan (and wifi is disabled) the on-board bluetooth seems to play okay together with the speaker.
I bought the bluetooth dongle because the audio was choppy when using the on-board bluetooth together with wifi.
So maybe thats just it..
The RPi 3B has to be connected to Lan and I have to use the on-board bluetooth.

Does anyone have experience on using wifi and bluetooth simultaniously on RPi 4?
Is it better than on RPi 3B?

**Comment by AndyFx12** [comment]
I don't have a solution - but I can confirm same behaviour with same BT- Dongle on my RPi3B. 
@rern: Do we have to deal with? Or is there any way to debug this?
Regards, Andy

**Reply by birkb85** [reply]
Thank you for your reply.
Im glad to know its not because I am doing something wrong.

**Comment by AndyFx12** [comment]
@rern 
<del>Issue seems to be rAudio related - my external BT-Dongle disconnects at once, if a track-change or refresh (like a new cover at webradio-playback) of the page occurs. If I start a long playlist, switch every client to off and don't touch any screen, my speaker-connection has no problem. Refresh the page from my smartphone disconnects the speaker at once.
TP-Link UB500 is a RTL8761B-based BT-Dongle. This dongle works with newer linux-kernels out of the box.
It would be really cool, if external BT-Dongles works like internal BT - working distance is nearly doubled compared to on-board BT.
@birkb85 
Maybe you can try, if your setup shows the same effect?</del>

EDIT: BT-Dongle disconnects on every action which needs some power from RPi3 - e. g. access via SCP.
dmesg shows "hci1: hardware error 0x03" - so I assume it's not an rAudio-issue ...

regards, Andy

**Reply by birkb85** [reply]
Thank you for the update!
I did not know about the dmesg command, I will also try that.

**Reply by birkb85** [reply]
Just tested today.

RPi connected through internal wifi.
Touch screen turned off.
Speaker connected to RPi through UB500 bluetooth dongle.
Started playing long play list.
Closed browser on my laptop so no device is using the rAudio browser.
The playlist plays for about 5 minutes and then the speaker disconnects.
It disconnects in the middle of a song.

with the dmesg command (sudo dmesg -L -H -T --follow) I get similar error:
[Fri Jun  2 08:10:45 2023] Bluetooth: hci0: hardware error 0x03
[Fri Jun  2 08:10:45 2023] WARN::dwc_otg_hcd_urb_dequeue:638: Timed out waiting for FSM NP transfer to complete on 3
